### PR TITLE
Added support for custom TRT plugin in trtexec

### PR DESCRIPTION
**Pull request by andi4191**
@rajeevsrao: Added support for external custom TensorRT plugin library.

Signed-off-by: andi4191 <a.dixit91@gmail.com>

### CODE REVIEW

**Comment by zeroepoch** [comment]
Wouldn't another way to accomplish this be using `LD_PRELOAD`?  This approach does seem cleaner though.

File: samples/opensource/trtexec/trtexec.cpp [hunk 1]
```diff
@@ -798,6 +816,17 @@ int main(int argc, char** argv)
 
     initLibNvInferPlugins(&gLogger.getTRTLogger(), "");
 
+#if !defined(_WIN32)
+    for (auto plugin : gParams.plugins)
+    {
+        void *dlhandle = dlopen(plugin.c_str(), RTLD_LAZY);
```
Comment (pranavm-nvidia) [inline samples/opensource/trtexec/trtexec.cpp]: We should create a cross-platform helper in `common.h` for loading libraries. I think the Windows equivalent is `LoadLibrary`

Comment (andi4191) [reply samples/opensource/trtexec/trtexec.cpp]: Do we support windows platform in OSS build yet?

**Comment by andi4191** [comment]
@zeroepoch : Does the LD_PRELOAD work for QNX too? It seems I am unable to run it with LD_PRELOAD

**Comment by zeroepoch** [comment]
> @zeroepoch : Does the LD_PRELOAD work for QNX too? It seems I am unable to run it with LD_PRELOAD

That's a good question.  I don't know if I've ever tried.  I maybe made an incorrect assumption that you were primarily working on solving the problem for desktop Linux.

**Comment by andi4191** [comment]
@pranavm-nvidia: I have incorporated your suggestion and created a wrapper API in common for loading library supporting multiple platforms. Request you for review comments.

File: samples/common/common.h [hunk 2]
```diff
@@ -771,4 +776,20 @@ inline std::ostream& operator<<(std::ostream& os, const nvinfer1::Dims& dims)
     return os << ")";
 }
 
+inline int loadLibrary(std::string libName)
```
Comment (pranavm-nvidia) [inline samples/common/common.h]: nit: `const std::string&`

File: samples/opensource/trtexec/trtexec.cpp [hunk 3]
```diff
@@ -798,6 +807,14 @@ int main(int argc, char** argv)
 
     initLibNvInferPlugins(&gLogger.getTRTLogger(), "");
 
+    for (auto plugin : gParams.plugins)
```
Comment (pranavm-nvidia) [inline samples/opensource/trtexec/trtexec.cpp]: nit: `const auto&`

File: samples/opensource/trtexec/trtexec.cpp (hunk 1 above)
Comment (andi4191) [reply samples/opensource/trtexec/trtexec.cpp]: fixed

**Pull request merged**